tests/functional/Teradata/ToStage/StageImportTest.php

```diff
@@ -123,4 +127,63 @@ public function testItWontFitIn(): void
             $this->getImportOptions()
         );
     }
+
+
+    public function testMoveDataFromAToBRequireSameTablesFailColumnNameMismatch(): void
+    {
+        $this->connection->executeQuery(
+            sprintf(
+                'CREATE MULTISET TABLE %s.%s
+            (
+            "id" INTEGER,
+    "first_name" VARCHAR(100),
+    "last_name" VARCHAR(100)
+    );',
+                TeradataQuote::quoteSingleIdentifier($this->getDestinationDbName()),
+                TeradataQuote::quoteSingleIdentifier('sourceTable')
+            )
+        );
+
+        $this->connection->executeQuery(
+            sprintf(
+                'CREATE MULTISET TABLE %s.%s
+            (
+            "id" INTEGER,
+    "first_name" VARCHAR(100),
+    "middle_name" VARCHAR(100),
+    "last_name" VARCHAR(100)
+    );',
+                TeradataQuote::quoteSingleIdentifier($this->getDestinationDbName()),
+                TeradataQuote::quoteSingleIdentifier('targetTable')
+            )
+        );
+
+        $importer = new ToStageImporter($this->connection);
+        $targetTableRef = new TeradataTableReflection(
+            $this->connection,
+            $this->getDestinationDbName(),
+            'targetTable'
+        );
+
+        $source = new Table(
+            $this->getDestinationDbName(),
+            'sourceTable',
+            ['id', 'first_name', 'last_name'],
+            []
+        );
+
+        $this->expectException(ColumnsMismatchException::class);
+        $this->expectExceptionMessage('Source destination columns name mismatch. "last_name"->"middle_name"');
```

Comment:
tady kontrolujes jenom nazvy sloupcu, ale mela by tu byt i kontrola typu?

Reply:
ne, kontroluje se tam i typ a a delka https://github.com/keboola/php-db-import-export/blob/master/src/Backend/Assert.php#L17

Reply:
jo, to vim, ze se to tam kontroluje, ale myslel jsem spis takovejhle test: https://github.com/keboola/php-db-import-export/blob/master/tests/unit/Backend/Synapse/AssertTest.php#L465
Ale je to uz pokryty v `AssertTest`, tak cajk.